### ISSUE TITLE
Spelling checker stops working after page changes

### ISSUE THREAD

**Issue by vossenjp** [zim-desktop-wiki/zim-desktop-wiki#2167]
**Describe the bug**
The spelling checker stops checking spelling after Zim page changes, on multiple versions of Zim, on two different machines.  This is very frustrating, because I'm a terrible speller and typist.

**To Reproduce**
1. Run Zim
2. Spell a word wrong, like foooooo, it should be flagged as spelled wrong 
3. Switch to a different (long) page
4. Spell a word wrong, like barrrrrrrr, it may or may not be flagged as spelled wrong 
5. Switch back to the original (long) page
6. The original misspelled word will probably still be flagged, so add a new misspelled work like bazzzzzz and it will not be flagged

I alluded to this in https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/1596#issuecomment-955581534 but never created a new bug until now.

**Expected behavior**
The spelling checker should Just Work and keep on working unless disabled.

**Work around**
F7 to turn off then on does NOT fix the problem. :-(  ONLY unchecking it in plugin prefs then re-checking it "fixes" the problem...until the next page change.

**Environment:**
 - Version: I don't recall when it started...after the switch to Python3 though.  It's been a while.  NOT a 0.75 issue, much older.  I was hoping 0.75 or my new faster H/W would help, but neither did.
 - OS: Various Linux Mint on 2 wildly different hosts
 - Language: English
 - Using: `gtkspellcheck` on both

**Additional context**
I'm not positive, but it seems to happen more on longer pages, possibly with lots of flagged (technical) words.  The 2 pages that most bother me are (`wc -l`):
* 3761 .../Zim/1_Work.txt
* 2680 .../Zim/1_Work/1_Con_Calls.txt

One of the affected nodes is a Mint-20.3 VM running in VBox-6.1.x running on an old Windows PC for $WORK, so it has 8G RAM but only 2 old, slow CPUs.  Data is also stored over a VBox "shared folder" (e.g., SMB) and all aspects of that are SLOW!  Zim is actually hard to use on this VM, because it's all so slow.  But I can't live without it, so...  The uncheck/recheck in plugins prefs takes at least 10 seconds to work, so that's painful too.  What I mean is, CTRL-, > Plugins tab: uncheck "Spell Checker," immediately re-check it, and hit OK.  For that to take effect and for the dialog to close and let me use Zim again is at least 10 seconds (can't check right now).

The other node was also quite old, until I replaced it a couple of months ago.  Now it has 32G RAM, 8x CPU, and NVME storage so it's really fast.  But this problem persists, which really surprised me (and is why I never filed a bug...I thought it was a time-out due to my old crappy stuff).  Here the uncheck/recheck in plugins prefs takes only ~3 seconds to get back to Zim and be ready for work, so that's livable.

Note even on the new PC, changing pages in Zim can take a few seconds, which is tedious.  That's because I have a good number of longer pages, 2,000-3,000 lines.  If I disable the spell checker (which again, I need), page changes are instant.  See #1497.

**Comment by vossenjp** [comment]
**tl;dr:** If Zim is very slow or if the spell checker is misbehaving, try removing `gtkspellcheck` and using `gtkspell` instead.  For Ubuntu, Mint, etc.:

* `sudo apt purge python3-enchant python3-gtkspellcheck`
* `sudo apt install gir1.2-gtkspell3-3.0 libgtkspell3-3-0`
* `sudo add-apt-repository ppa:jaap.karssenberg/zim && sudo apt update  && sudo apt install zim`

## Details:

I've had various problems with the spell checker (which I require :), e.g. #2167, #1497, and maybe #1596, which I believe all started with the move to Python3.  I've the same problems on multiple machines, running Mint-19.x, 20.x and 21.0, with various resources.  In particular, they persisted when I built a brand new 16 core Ryzen 7 which should not present any performance problems.  Related to performance, note I do keep Zim wiki files on a remote SSH FileSystem mount, but it's a local 1G LAN.

I had the "ah ha" moment when I build a brand new Mint-21.0 VM for work, installed Zim from PPA, and it Just Worked Great, with none of the problems.  When I looked into it, I saw "Edit > Preferences > Plugins > Spell Checker: gtkspell - OK (gtkspellcheck - Failed)".  Meanwhile, all the bad ones had "gtkspellcheck - OK" I assume for some historical reason or other.  When I switched a bad one from `gtkspellcheck` to `gtkspell` all the problems went away!

I did not specifically install either `gtkspellcheck` or `gtkspell` on this new VM, so I assume that `gtkspell` is (now?) the default somewhere.  So I am thinking that the move to Python3 somehow broke `gtkspellcheck` in a Zim context?  And maybe since `gtkspell` seems to be the default now, that's why most folks don't have this problem?

This also explains why turning spell checking off (#1497) "fixes" the problem.

**Comment by introt** [comment]
@vossenjp thanks for this information, could you open a ticket to include it in the docs?
